src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -450,7 +472,15 @@ private bool ProcessBlock(Block block, ChainedHeader header)
                 }
 
                 this.outpointsRepository.RecordRewindData(rewindData);
-                this.outpointsRepository.PurgeOldRewindData(this.consensusManager.Tip.Height - this.compactionTriggerDistance);
+
+                int purgeRewindDataThreshold = Math.Min(this.consensusManager.Tip.Height - this.compactionTriggerDistance, this.lastSavedHeight);
+
+                if (purgeRewindDataThreshold % 1000 == 0)
```

Comment:
A minor nitpick, the comment in the purge method is now inaccurate, as rewind data will now be getting purged in bulk rather than one block's data getting purged each block.